### PR TITLE
Cherry-pick 5.15 kernel changes to 6.6

### PR DESCRIPTION
**Pull request by rlmenge**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Align 3.0 with 2.0's kernel configs
Note that `CONFIG_IOMMU_SVA_LIB` was renamed `CONFIG_IOMMU_SVA` [[7ba564722d98e3e7bc3922ad4f2885ca0336674e](https://github.com/torvalds/linux/commit/7ba564722d98e3e7bc3922ad4f2885ca0336674e)], the `CONFIG_IOASID` was removed [[99b5726b44230329f35b4c4d7fe1577d4f4edb31](https://github.com/torvalds/linux/commit/99b5726b44230329f35b4c4d7fe1577d4f4edb31)], and I added the DEBUG option for QCOM 

###### Change Log  

- Cherry-pick 2.0 kernel config commits
  * Tue Nov 28 2023 Thien Trung Vuong <tvuong@microsoft.com>
    Enable CONFIG_BPF_LSM
  * Tue Nov 28 2023 Juan Camposeco <juanarturoc@microsoft.com>
    Enable CUSE module
  * Wed Nov 22 2023 David Daney <daviddaney@microsoft.com>
    Add IOMMU configs for aarch64

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: 470954
After rebase 
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=471434&view=results

### CODE REVIEW

**Review by jcamposeco — APPROVED** [review]

**Comment by Camelron** [comment]
Might as well cherry-pick the user events as well, yea? 

https://github.com/microsoft/CBL-Mariner/pull/7044

**Review by Camelron — APPROVED** [review]

**Comment by rlmenge** [comment]
> Might as well cherry-pick the user events as well, yea?
> 
> #7044

This one is already going into 3.0 :)

**Review by Camelron — APPROVED** [review]

File: SPECS/kernel/kernel.spec [hunk 1]
```diff
@@ -433,10 +433,12 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 - Add cpupower.service to kernel-tools
 - Enable user-based event tracing
 - Cherry-pick 2.0 kernel config commits
--  * Tue Nov 28 2023 Thien Trung Vuong <tvuong@microsoft.com>
--  - Enable CONFIG_BPF_LSM
--  * Tue Nov 28 2023 Juan Camposeco <juanarturoc@microsoft.com>
--  - Enable CUSE module
+  * Tue Nov 28 2023 Thien Trung Vuong <tvuong@microsoft.com>
```
Comment (christopherco) [inline SPECS/kernel/kernel.spec:436]: I don't believe this is valid `%changelog` syntax 
https://rpm-packaging-guide.github.io/#working-with-spec-files

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/kernel/kernel.spec (hunk 1 above)
Comment (christopherco) [reply SPECS/kernel/kernel.spec:436]: "Each item begins with a - character."

Comment (rlmenge) [reply SPECS/kernel/kernel.spec]: Adjusted each commit's changelog entry to better reflect expected format

**Review by trungams — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**